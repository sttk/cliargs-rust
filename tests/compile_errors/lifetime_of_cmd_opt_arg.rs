use cliargs::Cmd;

fn main() {
    fn returns_opt_arg() -> &'static str {
        let mut cmd =
            cliargs::Cmd::with_strings(["/path/to/app".to_string(), "--foo=bar".to_string()]);
        cmd.parse().unwrap();

        let opt_arg = cmd.opt_arg("foo").unwrap();
        println!("option arg (within the scope = {opt_arg:?}");

        opt_arg
    }

    let opt_arg = returns_opt_arg();
    println!("option arg (out of the scope) = {opt_arg:?}");
}
