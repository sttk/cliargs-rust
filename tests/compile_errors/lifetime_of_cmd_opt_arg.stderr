warning: unused import: `cliargs::Cmd`
 --> tests/compile_errors/lifetime_of_cmd_opt_arg.rs:1:5
  |
1 | use cliargs::Cmd;
  |     ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0515]: cannot return value referencing local variable `cmd`
  --> tests/compile_errors/lifetime_of_cmd_opt_arg.rs:12:9
   |
9  |         let opt_arg = cmd.opt_arg("foo").unwrap();
   |                       --- `cmd` is borrowed here
...
12 |         opt_arg
   |         ^^^^^^^ returns a value referencing data owned by the current function
